Debug errors in BadIP

nitin-techie opened the pull request:
BadIP was initially making decision based on just 1 category listed for the IOC. Also, the category was not displayed.
Customized the code to consider all the categories marked and display result that includes list of all the categories the IP was reported under.

## Description ##
BadIP was initially making decision based on just 1 category listed for the IOC. Also, the category was not displayed.
Customized the code to consider all the categories marked and display result that includes list of all the categories the IP was reported under.


## Does this fix a known existing bug under Issues? ##
Issue #56 

## Type of Change ##
Please delete any options that **do not** apply here:
- [ ] Bug Fix

## Any further info related to the addition ##
Provide any additional details that seem relevant here if necessary, otherwise either delete or leave this section blank.

nitin-techie commented:
@TheresAFewConors -- Pls review and if you're okay with the changes, You can merge the latest commit of Sooty.py.

nitin-techie commented on Sooty.py:
```diff
@@ -426,10 +426,18 @@ def repChecker():
-
-                sc = result['Score']['ssh']
+                sc = 0
+                for each in result['Categories']:
+                    temp = each
+                    sc += result['Score']['temp']
```
Calculate the score for all the categories it was reported under and sum it all.

nitin-techie commented on Sooty.py:
```diff
@@ -426,10 +426,18 @@ def repChecker():
                 print("  Score: " + str(sc))
+                if sc == 0:
+                    print('  Score 0 indicates, this might be a false positive')
+                print("\n IP reported in following Categories :\n")
+                for each in result['Categories']:
+                    print(" *. " + each)        
```
Print all the categories name it was reported under

TheresAFewConors reviewed (commented):
looks like there's an error on the sc value - looking into it

TheresAFewConors commented:
looks like on BadIP's all the scores are set to 0 from the small subset I looked at. Rewrote a section to skip the BadIP value, (Total Reports provides a good metric), and also added the timestamps of last report.

nitin-techie commented:
> looks like on BadIP's all the scores are set to 0 from the small subset I looked at. Rewrote a section to skip the BadIP value, (Total Reports provides a good metric), and also added the timestamps of last report.

Some scores are definitely greater than 0.
Besides, if you would'd like to skip that part. I am okay with it.

TheresAFewConors commented:
> 
> 
> > looks like on BadIP's all the scores are set to 0 from the small subset I looked at. Rewrote a section to skip the BadIP value, (Total Reports provides a good metric), and also added the timestamps of last report.
> 
> Some scores are definitely greater than 0.
> Besides, if you would'd like to skip that part. I am okay with it.

That is quite likely, I only checked 5 or 6 of the more recent reports. 

I just pushed the fix, I like what you done with the specific categories being reported as well, very informative and useful information, I just added the associated timestamp for each category too.

nitin-techie commented:
> > > looks like on BadIP's all the scores are set to 0 from the small subset I looked at. Rewrote a section to skip the BadIP value, (Total Reports provides a good metric), and also added the timestamps of last report.
> > 
> > 
> > Some scores are definitely greater than 0.
> > Besides, if you would'd like to skip that part. I am okay with it.
> 
> That is quite likely, I only checked 5 or 6 of the more recent reports.
> 
> I just pushed the fix, I like what you done with the specific categories being reported as well, very informative and useful information, I just added the associated timestamp for each category too.

#

> > > looks like on BadIP's all the scores are set to 0 from the small subset I looked at. Rewrote a section to skip the BadIP value, (Total Reports provides a good metric), and also added the timestamps of last report.
> > 
> > 
> > Some scores are definitely greater than 0.
> > Besides, if you would'd like to skip that part. I am okay with it.
> 
> That is quite likely, I only checked 5 or 6 of the more recent reports.
> 
> I just pushed the fix, I like what you done with the specific categories being reported as well, very informative and useful information, I just added the associated timestamp for each category too.

Thanks.
Can i be added to the contributors list ?

TheresAFewConors commented:
> Thanks.
> Can i be added to the contributors list ?

Of course, I am just waiting for the automated checks to complete first to make sure there are no issues. I will update the list then afterwards.

nitin-techie commented:
> > Thanks.
> > Can i be added to the contributors list ?
> 
> Of course, I am just waiting for the automated checks to complete first to make sure there are no issues. I will update the list then afterwards.

- [ ] @TheresAFewConors -- Thanks, i am gonna be opening a couple of more issues, which i have already sorted in my local file.
